### ISSUE TITLE
New contributor documentation improvements for repo

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#24575]
This issue is used to track documentation improvements we're working on collectively for this repo. Check off each item as you complete it and add more.

- [ ] Clarify what workloads need to be installed with VS
- [ ] Document how to run test cases per project and how to run Selenium tests
- [ ] Update ordering on installation setup so that it flows from checkout source -> install dependencies -> build project -> open files -> make edits -> run tests
- [ ] Notify users about issues with building on Windows with long filenames or Unicode characters in name
- [ ] Add note about needing US culture for tests

**Comment by Pilchie** [comment]
Not exactly new contributor related, but another thing that could be made more clear is that if you're building a tag for a specific release, you might need to edit the `NuGet.config` file to remove internal, authenticated NuGet feeds (the ones that start with `darc-int`).

**Comment by SQL-MisterMagoo** [comment]
Hi, I'm not sure how much use this is, but I was having problems with the Restore.cmd erroring. 

> TLDR; If the Restore or Build hangs and you Break into it, it might be best to answer "N" to terminating the batch.
> ...or it might have been a coincidence...

I was seeing this error
> .dotnet\sdk\5.0.100-rc.1.20379.10\NuGet.targets(128,5): error : Unable to load the service index for source  https://api.nuget.org/v3/index.json.
> ..but that URL loads fine in a browser

I tried the same thing over a few days and it always errored the same way and then the batch process would just hang with no apparent CPU or network activity.

Suddenly yesterday it started working again and the only thing I did differently was to answer "N" to the question "do you want to terminate the batch process?" when it had hung again and I had pressed ctrl-c to break. 

After I answered "N" to that question, the batch process ended within a second anyway, but the next run succeeded and subsequent runs also now work.

**Comment by dougbu** [comment]
@SQL-MisterMagoo sounds like you hit a transient issue with the NuGet.org site rather than a `darc-int-...` feed that wasn't available outside our internal build infrastructure.

> you might need to edit the NuGet.config file

@Pilchie, `darc-int-...` feeds remain an issue for older tags. Problem shouldn't exist for future tags because those feeds will by default be disabled. Look forward to our next MSRC release 😺 

Should we document the problems for earlier tags versus expecting people to mostly use the latest❔

**Comment by dougbu** [comment]
Including this in the DoI project for discussion in that context

**Comment by javiercn** [comment]
@captainsafia are we done with this one?

**Comment by wtgodbe** [comment]
Going to close this one as the state info is probably stale at this point - @captainsafia feel free to re-open/open a new one if this is something we still need to improve
